trucks.js

```diff
@@ -501,4 +503,82 @@ var foodTrucks = [
 		Facebook: 'https://www.facebook.com/fulltilticecreams',
 		Twitter: 'https://twitter.com/fticecream'
 	}
-];
\ No newline at end of file
+];
+
+// this module  supports the following methods:
+// getTrucks() - return all trucks
+// getTruck(name) - return the truck object matching 'name'
+// getFoodTypes() - return unique list of all associated food types (underscore has a function to help)
+// filterByDay(day) - return trucks with 'day' in schedule (use your filterByDay function from Module 3 homework)
+// filterByFoodType(foodType) - return trucks with associated 'foodType'
+
+function getTrucks() {
+    return foodTrucks;
+}
+
+function getTruck(name) {
+    //update this to put the single truck into a form
+    //that truck can them be updated
+	return _.filter(foodTrucks, function (truck) {
+		return truck.name === name;
+	});
+}
+
+function getFoodTypes() {
+    
+	return _.uniq(_.flatten(_.pluck(foodTrucks, 'type')));
+}
+
+function filterByDay(dayName) { 
+
+    return _.filter(foodTrucks, function (truck) { 
+	    return _.any(truck.schedule, function (day) { 
+		    return dayName === day;
+	     }); 
+    }); 
+}
+
+function filterByFoodType(foodType) {
```

Comment:
You had the right idea with using _.filter. What the _.where function is looking for is objects in the list that have a property type that matches the value, but the property type in this case is an array of values. It will never match exactly.